### PR TITLE
Add support for Access Entry type

### LINKED ISSUE #7446: [Bug] Access Entries for Self-Managed Nodes not created by eksctl not working
### What were you trying to accomplish?
Create an access entry for an EC2_LINUX IAM role for Karpenter to use.

### What happened?

When attempting to create an Access Entry for an EC2 Linux self-managed node group used by Karpenter via the config file, eksctl is expecting a policy ARN to be passed. The documentation for creating these types of access entries seems to be lacking, or perhaps eksctl currently only supports creation of these entries for self-managed node groups that eksctl creates (as opposed to compatibility for external self-managed node groups, such as Karpenter nodes).

### How to reproduce it?

First tried to create the entry without specifying the type. This won't work because it will try to create as a standard entry, which doesn't allow `system:*` to be passed in the `kubernetesGroups`.

```
accessConfig:
  authenticationMode: API_AND_CONFIG_MAP
  accessEntries:
    - principalARN: arn:aws:iam::11111111111:role/KarpenterNodeInstanceProfile
      kubernetesUsername: system:node:{{EC2PrivateDNSName}}
      kubernetesGroups:
        - system:bootstrappers
        - system:nodes
```

`The kubernetes group name system:nodes is invalid, it cannot start with system`

Then, tried passing the type as EC2_LINUX, but eksctl is expecting a policyARN to be passed. 

```
accessConfig:
  authenticationMode: API_AND_CONFIG_MAP
  accessEntries:
    - principalARN: arn:aws:iam::11111111111:role/KarpenterNodeInstanceProfile
      kubernetesUsername: system:node:{{EC2PrivateDNSName}}
      kubernetesGroups:
        - system:bootstrappers
        - system:nodes
      accessPolicies:
        - accessScope:
            type: EC2_LINUX
```

`Error: accessEntries[0].policyARN must be set to a valid AWS ARN`

I believe my second attempt is the correct approach, and that eksctl should not be requiring a policyARN to be passed for this type. 

**Logs**
`Error: accessEntries[0].policyARN must be set to a valid AWS ARN`

**Versions**

```
eksctl version: 0.167.0-dev+4280da7cd.2023-12-22T09:42:31Z
kubectl version: v1.29.0
OS: darwin
```

### PR DESCRIPTION
**Pull request by yuxiang-zhang**
### Description

Resolves #7446

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: examples/40-access-entries.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ kind: ClusterConfig
 metadata:
   name: access-entries-cluster
   region: us-west-2
-  version: '1.25'
```
Comment (a-hilaly) [inline examples/40-access-entries.yaml:9]: Did this example work before? or it's not being tested at all?

File: pkg/apis/eksctl.io/v1alpha5/access_entry.go [hunk 2]
```diff
@@ -6,16 +6,20 @@ import (
...
 
 	"github.com/aws/aws-sdk-go-v2/aws/arn"
+	ekstypes "github.com/aws/aws-sdk-go-v2/service/eks/types"
 )
 
 // AccessEntry represents an access entry for managing access to a cluster.
 type AccessEntry struct {
+	// existing IAM principal ARN to associate with an access entry
 	PrincipalARN ARN `json:"principalARN"`
-	// +optional
```
Comment (a-hilaly) [inline pkg/apis/eksctl.io/v1alpha5/access_entry.go:14]: Not very sure here but i think `// +optional` are supposed to be a one line comment. They are parsed by controller-gen if i'm not mistaken..

Comment (a-hilaly) [reply pkg/apis/eksctl.io/v1alpha5/access_entry.go:14]: See https://github.com/kubernetes-sigs/controller-tools/pull/211/files#diff-3b88366c37358db48ac32a74f079eb2d1c89727c3cd99ddb552908a05794c94fR66-R70

Comment (a-hilaly) [reply pkg/apis/eksctl.io/v1alpha5/access_entry.go:14]: Do we use `controller-gen` to generate CRDs for eksctl?

File: examples/40-access-entries.yaml (hunk 1 above)
Comment (yuxiang-zhang) [reply examples/40-access-entries.yaml:9]: Yeah it worked with non `STANDARD` types. I just updated to show the new fields in use

**Review by a-hilaly — DISMISSED** [review]
Thank you @yuxiang-zhang ! 👍

**Review by a-hilaly — APPROVED** [review]

**Pull request merged**